lib/tracer/tracer.js

```diff
@@ -210,21 +215,47 @@ Tracer.prototype.buildEvent = function (event, depth = 0, trace) {
   return builtEvent;
 };
 
+Tracer.prototype._buildMetrics = function ({ events, traceInfo, metrics, onBuild }) {
+  for (let i = 0; i < events.length; i++) {
+    let prevEvent = i > 0 ? events[i - 1] : null;
+    let event = events[i];
+
+    if ([EventType.START, EventType.COMPLETE].includes(event.type)) continue;
+
+    if (!event.endAt) {
+      console.error('Monti APM: no end event for type: ', event.type);
+      continue;
+    }
+
+    const computeTime = prevEvent ? event.at - prevEvent.endAt : 0;
+
+    let builtEvent = this.buildEvent(event, 0, traceInfo);
+
+    if (onBuild) onBuild({ builtEvent, computeTime });
+
+    metrics[event.type] = metrics[event.type] || 0;
+    metrics[event.type] += builtEvent[1];
+
+    if (event.type === EventType.CUSTOM && isFilledArray(event.nested)) {
+      this._buildMetrics({ events: event.nested, traceInfo, metrics });
```

Comment:
The `metrics` object shouldn't have a custom property.

Reply:
Oh, that's right. Thanks